src/devices/Board/Board.cs

```diff
@@ -0,0 +1,692 @@
+        /// </summary>
+        protected int RemapPin(int pin, PinNumberingScheme providedScheme)
+        {
+            return ConvertPinNumber(pin, providedScheme, PinNumberingScheme.Logical);
+        }
+
+        /// <summary>
+        /// Converts a set of pins to the logical scheme
+        /// </summary>
+        protected int[]? RemapPins(int[] pins, PinNumberingScheme providedScheme)
+        {
+            if (pins == null)
+            {
+                return null;
+            }
+
+            if (providedScheme == PinNumberingScheme.Logical)
+            {
+                return pins;
+            }
+
+            int[] newPins = new int[pins.Length];
+            for (int i = 0; i < pins.Length; i++)
+            {
+                newPins[i] = ConvertPinNumber(pins[i], providedScheme, PinNumberingScheme.Logical);
+            }
+
+            return newPins;
+        }
+
+        private bool UsageCanSharePins(PinUsage usage)
+        {
+            switch (usage)
+            {
+                // Several SPI devices can share the same Pins (because we're allocating devices, not buses)
+                case PinUsage.Spi:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Create an instance of the best possible board abstraction.
+        /// </summary>
+        /// <param name="defaultNumberingScheme">The default pin numbering scheme the board should use</param>
+        /// <returns>A board instance, to be used across the application</returns>
+        /// <exception cref="PlatformNotSupportedException">The board could not be identified</exception>
+        /// <remarks>The detection concept should be refined, but this requires a public detection api</remarks>
+        public static Board Create(PinNumberingScheme defaultNumberingScheme = PinNumberingScheme.Logical)
+        {
+            Board? board = null;
+            try
+            {
+                board = new RaspberryPiBoard(defaultNumberingScheme);
+                board.Initialize();
+            }
+            catch (Exception x) when ((x is NotSupportedException) || (x is IOException))
+            {
+                board?.Dispose();
+                board = null;
+            }
+
+            if (board != null)
+            {
+                return board;
+            }
+
+            try
+            {
+                board = new GenericBoard(defaultNumberingScheme);
+                board.Initialize();
+            }
+            catch (Exception x) when ((x is NotSupportedException) || (x is IOException))
+            {
+                board?.Dispose();
+                board = null;
+            }
+
+            if (board != null)
+            {
+                return board;
+            }
+
+            throw new PlatformNotSupportedException("Could not find a matching board driver for this hardware");
```

Comment:
Is there a reason why we don't try as well with CustomBoard?

Reply:
Because the CustomBoard _requires_ custom drivers. It's "default" implementation would be equivalent to the GenericBoard.